python-primitiv/primitiv/_operator.pyx

```diff
@@ -16,28 +16,27 @@ import numpy as np
 class _operators:
 
     @staticmethod
-    def input(data, shape = None, _Device device = None, _Graph g = None):
-        cdef vector[float] data_vector
+    def raw_input(shape, vector[float] data, _Device device = None, _Graph g = None):
         if device is None:
             device = _Device.get_default()
+        if g is not None:
+            return wrapNode(Node_input_vector(normShape(shape).wrapped, data, device.wrapped[0], g.wrapped[0]))
+        else:
+            return wrapNode(Node_input_vector(normShape(shape).wrapped, data, device.wrapped[0]))
+
+    # NOTE(vbkaisetsu)
+    # This function takes an np.ndarray or a list of np.ndarray
+    # instead of a vector.
+    @staticmethod
+    def input(data, _Device device = None, _Graph g = None):
         if isinstance(data, np.ndarray):
             data = [data]
-        elif not isinstance(data, list):
-            raise TypeError("Argument 'data' has incorrect type (list or numpy.ndarray)")
         if len(data) == 0:
```

Comment:
Let's put this condition at the top of this function. `len(ndarray)` returns the first dimension, and we can avoid empty `ndarray`s by using the same condition.